src/nvim/screen.c

```diff
@@ -1948,6 +1948,29 @@ static inline void provider_err_virt_text(linenr_T lnum, char *err)
   decor_add_ephemeral(lnum-1, 0, lnum-1, 0, &err_decor);
 }
 
+/// Draw the line number (empty space after wrapping).
+static inline void draw_line_number(win_T *wp, linenr_T lnum, char_u *extra)
+{
+  long num;
+  char *fmt = "%*ld ";
+
+  if (wp->w_p_nu && !wp->w_p_rnu) {
+    // 'number' + 'norelativenumber'
+    num = (long)lnum;
+  } else {
+    // 'relativenumber', don't use negative numbers
+    num = labs((long)get_cursor_rel_lnum(wp, lnum));
+    if (num == 0 && wp->w_p_nu && wp->w_p_rnu) {
+      // 'number' + 'relativenumber'
+      num = lnum;
+      fmt = "%-*ld ";
+    }
+  }
+
+  snprintf((char *)(extra), sizeof(extra),
```

Comment:
Using `sizeof(extra)` here does not work either. You have to pass `sizeof(extra)` from `win_line()`.

Reply:
Sorry of course.

I'm a bit concerned this didn't cause a lot more test failures.